### PR TITLE
Feature/ssp 80/floorchooser

### PR DESCRIPTION
**Pull request by ndmytro**
Floor chooser implementation for plan viewer application.

### CODE REVIEW

**Comment by ArtemSokoliuk** [comment]
Don't forget to squash commits.

File: SSP/plan_editor/urls.py [hunk 1]
```diff
@@ -1,8 +1,7 @@
-# coding: utf-8
 from django.conf.urls import patterns, url
 
 __author__ = 'mironnn'
 
-urlpatterns = patterns('SSP.plan_editor.views',
+urlpatterns = patterns('plan_editor.views',
                        url(r'^$', 'show_map', name='show_map'),
                        )
```
Comment (Fedorof) [inline SSP/plan_editor/urls.py]: Newline

File: SSP/plan_viewer/admin.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+from django.contrib import admin
```
Comment (Fedorof) [inline SSP/plan_viewer/admin.py]: Do you really need this file?

File: SSP/plan_viewer/models.py [hunk 3]
```diff
@@ -0,0 +1,3 @@
+from django.db import models
```
Comment (Fedorof) [inline SSP/plan_viewer/models.py]: Why do you need this?

File: SSP/plan_viewer/templates/plan_viewer.html [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    <link href="{% static 'css/vendor/leaflet.draw.css' %}" rel="stylesheet" type="text/css" />
+    <link href="{% static 'css/vendor/leaflet.toolbar.css' %}" rel="stylesheet" />
+{% endblock %}
+
+
+{% block javascripts %}
+    {{ block.super }}
+    <script src="{% static 'js/vendor/leaflet.js' %}"></script>
+    <script src="{% static 'js/vendor/leaflet.draw.js' %}"></script>
+    <script src="{% static 'js/vendor/leaflet.toolbar.js'%}"></script>
```
Comment (Fedorof) [inline SSP/plan_viewer/templates/plan_viewer.html]: Correct me if I'm wrong - do we really need leaflet for plan *viewer*?

File: SSP/plan_viewer/tests.py [hunk 5]
```diff
@@ -0,0 +1,3 @@
+from django.test import TestCase
```
Comment (Fedorof) [inline SSP/plan_viewer/tests.py]: Another empty file.

File: SSP/plan_viewer/urls.py [hunk 6]
```diff
@@ -0,0 +1,6 @@
+__author__ = 'dn'
```
Comment (Fedorof) [inline SSP/plan_viewer/urls.py]: Remove this line, please.

File: SSP/ssp_project/urls.py [hunk 7]
```diff
@@ -18,7 +18,7 @@
 
 urlpatterns = [
     url(r'^admin/', include(admin.site.urls)),
-    url(r'^$', 'SSP.plan_editor.views.index', name='root_of_project'),
-
-    url(r'^edit_plan/', include('SSP.plan_editor.urls')),
+    url(r'^$', 'plan_editor.views.index', name='root_of_project'),
+    url(r'^edit_plan/', include('plan_editor.urls')),
+    url(r'^plan_viewer/', include('plan_viewer.urls')),
```
Comment (Fedorof) [inline SSP/ssp_project/urls.py]: I suppose this should be a main page.

**Comment by Fedorof** [comment]
@ITsvetkoFF , you knows JS better than me

File: SSP/plan_viewer/urls.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
+from django.conf.urls import url
+
+urlpatterns = [
+    url(r'^$', 'plan_viewer.views.main'),
+    ]
```
Comment (Fedorof) [inline SSP/plan_viewer/urls.py]: Newline

File: SSP/plan_viewer/views.py [hunk 9]
```diff
@@ -0,0 +1,6 @@
+from django.shortcuts import render
+
+# Create your views here.
```
Comment (Fedorof) [inline SSP/plan_viewer/views.py]: Remove this comment.

File: SSP/plan_viewer/views.py [hunk 10]
```diff
@@ -0,0 +1,6 @@
...
+
+def main(request):
+    return render(request, "plan_viewer.html")
```
Comment (Fedorof) [inline SSP/plan_viewer/views.py]: It would be nice to have some docstring here

File: SSP/ssp_project/urls.py [hunk 11]
```diff
@@ -20,4 +20,5 @@
     url(r'^admin/', include(admin.site.urls)),
     url(r'^$', 'plan_editor.views.index', name='root_of_project'),
     url(r'^edit_plan/', include('plan_editor.urls')),
+    url(r'^plan_viewer/', include('plan_viewer.urls')),
```
Comment (Fedorof) [inline SSP/ssp_project/urls.py]: Perhaps it should be root

File: SSP/static/js/plan_viewer.js [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+    var floor1 = L.ToolbarAction.extend({
+            options: {
+                toolbarIcon: {
+                    html: '1',
+                    tooltip: 'Choose floor #1'
+                }
+            },
+            addHooks: function () {
+                alert('No map for floor 1');
```
Comment (Fedorof) [inline SSP/static/js/plan_viewer.js]: Perhaps just a text message? Alerts are annoying.

File: SSP/static/js/plan_viewer.js [hunk 13]
```diff
@@ -0,0 +1,106 @@
+$(document).ready(function() {
+
+    var map = L.map('map', {crs: L.CRS.Simple}).setView([12, 10], 5);
+    var imageUrl = 'https://s3.amazonaws.com/uploads.hipchat.com/466087/2978340/1nc1yFkOs8gkHmS/Kyiv1%20poverh%202-3-page-001.jpg',
+    imageBounds = [[0, 0], [24.81, 17.54]];
+    var floorMap = L.imageOverlay(imageUrl, imageBounds);
+    map.addLayer(floorMap);
+
+
+    var floor1 = L.ToolbarAction.extend({
```
Comment (Fedorof) [inline SSP/static/js/plan_viewer.js]: Is there a way to have floors constructions not hardcoded? Store the information about floors in json and use it do build the floors (you will get that json from the server in the future)

**Comment by Fedorof** [comment]
And having some tests would be nice

File: SSP/plan_editor/views.py [hunk 14]
```diff
@@ -4,11 +4,9 @@
...
 __author__ = 'mironnn'
 
-
 def index(request):
     return render(request, "index.html")
 
-
 def show_map(request):
     return render(request, "office_map.html")
     # return HttpResponse('ok')
```
Comment (Fedorof) [inline SSP/plan_editor/views.py]: What is this? Do you really need it?

File: SSP/plan_editor/views.py [hunk 15]
```diff
@@ -4,11 +4,9 @@
 
 __author__ = 'mironnn'
```
Comment (Fedorof) [inline SSP/plan_editor/views.py]: I would remove it.

File: SSP/plan_editor/views.py [hunk 16]
```diff
@@ -4,11 +4,9 @@
...
 
-
 def index(request):
```
Comment (Fedorof) [inline SSP/plan_editor/views.py]: What about docstring?

File: SSP/plan_editor/views.py [hunk 17]
```diff
@@ -4,11 +4,9 @@
...
     return render(request, "index.html")
 
-
 def show_map(request):
```
Comment (Fedorof) [inline SSP/plan_editor/views.py]: Docstring

File: SSP/plan_viewer/templates/plan_viewer.html [hunk 18]
```diff
@@ -0,0 +1,29 @@
...
+{% endblock %}
+
+
+{% block content %}
+
+    <div id="map" style="height: 700px; width: auto">
+
+    </div>
+
+{% endblock content %}
```
Comment (Fedorof) [inline SSP/plan_viewer/templates/plan_viewer.html]: Newline

File: SSP/ssp_project/settings.py [hunk 19]
```diff
@@ -38,6 +38,7 @@
     'django.contrib.messages',
     'django.contrib.staticfiles',
     'plan_editor',
+    'plan_viewer',
```
Comment (Fedorof) [inline SSP/ssp_project/settings.py]: Did we agreed about another app name?

File: SSP/ssp_project/urls.py [hunk 20]
```diff
@@ -18,6 +18,7 @@
 
 urlpatterns = [
     url(r'^admin/', include(admin.site.urls)),
-    url(r'^$', 'plan_editor.views.index', name='root_of_project'),
+    url(r'^$', 'plan_editor.views.index'),
```
Comment (Fedorof) [inline SSP/ssp_project/urls.py]: Perhaps "^$" should belong to the viewer

File: SSP/static/js/plan_viewer.js [hunk 21]
```diff
@@ -0,0 +1,58 @@
...
+            L.ToolbarAction.extend({
+            options: {
+                toolbarIcon: {
+                    html: floor.number,
+                    tooltip: "Choose floor #" + floor.number.toString()
+                }
+            },
+            addHooks: function () {
+                map.removeLayer(floorMap);
+                floorMap = L.imageOverlay(floor.image, [[0, 0], [24.81, 17.54]]);
```
Comment (Fedorof) [inline SSP/static/js/plan_viewer.js]: `[[0, 0], [24.81, 17.54]]` - what is the meaning of this data? Should it go to `floors_array`? You use it twice.

File: SSP/static/js/plan_viewer.js [hunk 22]
```diff
@@ -0,0 +1,58 @@
+$(document).ready(function() {
+
+    floors_array = [
```
Comment (Fedorof) [inline SSP/static/js/plan_viewer.js]: Perhaps `var floors_array`?

http://stackoverflow.com/questions/1470488/what-is-the-function-of-the-var-keyword-and-when-to-use-it-or-omit-it

File: SSP/static/js/plan_viewer.js [hunk 23]
```diff
@@ -0,0 +1,58 @@
...
+
+    var map = L.map('map', {crs: L.CRS.Simple, minZoom: 5, maxZoom: 7}).setView([12, 10], 5);
+    var imageUrl = floors_array[0].image;
+    imageBounds = [[0, 0], [24.81, 17.54]];
+    var floorMap = L.imageOverlay(imageUrl, imageBounds);
+    map.addLayer(floorMap);
+
+    new L.Toolbar.Control({
+    position: 'topleft',
+    actions: actions_array
```
Comment (Fedorof) [inline SSP/static/js/plan_viewer.js]: Please, add some indent.

File: SSP/static/js/plan_viewer.js [hunk 24]
```diff
@@ -0,0 +1,58 @@
...
+    }).addTo(map);
+})
+
+
```
Comment (Fedorof) [inline SSP/static/js/plan_viewer.js]: Hawing some tests for this functionality would be (awesome)

File: SSP/static/js/plan_viewer.js (hunk 22 above)
Comment (ndmytro) [inline SSP/static/js/plan_viewer.js]: Thank you for your advice! I have checked the link, it's definitely a basic thing everyone should know=)

File: SSP/ssp_project/settings.py (hunk 19 above)
Comment (ndmytro) [inline SSP/ssp_project/settings.py]: Yep, "Carcassonne" is the new one=)

File: SSP/plan_editor/views.py (hunk 17 above)
Comment (ndmytro) [inline SSP/plan_editor/views.py]: May I write a docstring for someone else's code?

File: SSP/static/js/plan_viewer.js (hunk 24 above)
Comment (ndmytro) [inline SSP/static/js/plan_viewer.js]: Probably the testing of this code will just test what's already tested - LeafletJS. Am I right?

File: SSP/static/js/plan_viewer.js (hunk 23 above)
Comment (ndmytro) [inline SSP/static/js/plan_viewer.js]: Roger that!

File: SSP/plan_editor/views.py (hunk 14 above)
Comment (ndmytro) [inline SSP/plan_editor/views.py]: I'm sure I don't need it at all, but it's not my code. Should I edit it?

File: SSP/ssp_project/urls.py (hunk 20 above)
Comment (ndmytro) [inline SSP/ssp_project/urls.py]: Will comply.

File: SSP/static/js/plan_viewer.js (hunk 21 above)
Comment (ndmytro) [inline SSP/static/js/plan_viewer.js]: These are the coordinates of the lower left and upper right corner of the image respectively. At the current moment this data cannot be stored elsewhere.

**Pull request closed without merging**